contractcourt/chain_arbitrator.go

```diff
@@ -763,20 +765,75 @@ func (c *ChainArbitrator) Start() error {
 
 	// Launch all the goroutines for each arbitrator so they can carry out
 	// their duties.
+	// Set a timeout for the group of goroutines. Maybe have different
+	// timeouts for itests and normal operations.
+	ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
+
+	// Create an errgroup with the context
+	errGroup, ctx := errgroup.WithContext(ctx)
+
 	for _, arbitrator := range c.activeChannels {
 		startState, ok := startStates[arbitrator.cfg.ChanPoint]
 		if !ok {
 			stopAndLog()
+			// In case we encounter an error we need to cancel the
+			// context to ensure all goroutines are cleaned up.
+			cancel()
 			return fmt.Errorf("arbitrator: %v has no start state",
 				arbitrator.cfg.ChanPoint)
 		}
 
-		if err := arbitrator.Start(startState); err != nil {
-			stopAndLog()
-			return err
-		}
+		errGroup.Go(func() error {
+			// Create buffered error channel for this specific
+			// arbitrator
+			errChan := make(chan error, 1)
+
+			// Start arbitrator in a separate goroutine
+			go func() {
+				errChan <- arbitrator.Start(startState)
```

Comment:
Isn't the whole goal of the `errGroup` that we have a context that is canceled when an error occurs?
But now we're starting a new goroutine inside the err group goroutine just so we can abort `Start()`?
This will work I think. But perhaps another possible approach would be to pass the cancellable context into `Start()` and abort on context cancel there?
Otherwise we'd kind of kill/abandon the goroutines spawned in `Start()` on shutdown.

Reply:
very good idea

Reply:
Also why isn't `arb.Start()` returned directly as the error here? That would actually use the `errgroup` features. 

Zooming out, what we want here is that `chanAbr.Start()` actually doesn't block, but AFAICT, it'll wait for all the goroutines to start below, which can still enter the deadlock scenario we were trying to resolve.

Reply:
not sure if I undestand your question, but I introduced the errGroup and this goroutine, because otherwise I cannot fail the goroutines as soon as a goroutines fails with an error. The normal errGroup waits until all goroutines are done which would bring us into the deadlock again. but yeah not really necessary wich the new appraoch.